vnext/Microsoft.ReactNative/ReactNativeIsland.idl

```diff
@@ -98,6 +98,9 @@ namespace Microsoft.ReactNative
     DOC_STRING("ScaleFactor for this windows (DPI/96)")
     Single ScaleFactor {get; set;};
 
+    DOC_STRING("Adds CompositionEventHandler to a ReactNativeIsland that doesn't have it's own RootView")
+    void AddFragmentCompositionEventHandler(Microsoft.ReactNative.IReactContext context, Microsoft.ReactNative.ComponentView componentView);
```

Comment:
From the end user point of view, they can either set a ReactViewHost on a ReactNativeIsland, OR they can call this method.  I wonder if we can make that clearer in the API somehow.

Part of me wonders if the functionality of these two cases will diverge over time, such that having them be the same type may be annoying long term.

Reply:
Main reason we need to think more about this is that this part of the change affects the public interface, so its more important to think through.
 
One improvement would be to use a new constructor that takes the react context + component view.
Then we should probably add some error handling if the user tries to set ReactViewHost on a ReactNativeIsland that was constructed as a fragment.

I know I said we could look at the layout part after this checkin, but it might be worth doing now to see how much that splits the implementation - which might help us work out if these should be separate classes or how else it might affect the public interface.

Reply:
I went ahead and tried to implement layout! It works similar to Popout/Flyout in fabric where you set a width/height to a view inside of the component. There's still some funky parts (ie setting flex/padding on the first View has some not so desirable affect) but I believe it should be enough for the first part. Let me know what you think though!

Since this layout part is inside the ModalComponentView, I was able to not have to change ReactNativeIsland. I went ahead and added a new Constructor to the public interfaces and added a check for setting a ReactViewHost as a fragment.